bindings/cs/rl.net/RLException.cs

```diff
@@ -0,0 +1,45 @@
+using System;
+using System.Runtime.Serialization;
+using System.Security.Permissions;
+
+namespace Rl.Net
+{
+    [Serializable]
+    public class RLException : Exception
+    {
+        private const string ErrorCodeSerializationKey = nameof(RLException.ErrorCode);
+
+        public int ErrorCode
+        {
+            get;
+            set;
+        } = -1;
+
+        public RLException(ApiStatus status) : base(status.ErrorMessage)
+        {
+            this.ErrorCode = status.ErrorCode;
+        }
+
+        public RLException(ApiStatus status, Exception inner) : base(status.ErrorMessage, inner)
+        {
+            this.ErrorCode = status.ErrorCode;
+        }
+
+        protected RLException(SerializationInfo info, StreamingContext context) : base(info, context)
+        {
+            this.ErrorCode = info.GetInt32(ErrorCodeSerializationKey);
+        }
+
+        [SecurityPermission(SecurityAction.Demand, SerializationFormatter = true)]
```

Comment:
Do we still need this stuff?

Reply:
If we want to support scenarios like remoting (or serializing for some other reason) exceptions, we need this. Is there a good reason to remove it?

Reply:
dotnet code doesn't support remoting, but has most of the serialization stack.
It's harmless to have it, OTOH.

Reply:
Fair enough. I'll keep this since we technically do support .Net Framework.